sympy/calculus/finite_diff.py

```diff
@@ -349,6 +349,14 @@ def _as_finite_diff(derivative, points=1, x0=None, wrt=None):
 (-sqrt(2)*h + E*h)/(2*h))*f(-h + x)/(h + E*h) + \
 (-(h + sqrt(2)*h)/(2*h) + (-sqrt(2)*h + E*h)/(2*h))*f(h + x)/(-h + E*h)
 
+    To approximate ``derivative`` around ``x0`` using a non-equidistant
+    spacing step, the algorithm supports assignment of undefined
+    functions to ``points``:
+
+    >>> dx = Function('dx')
+    >>> as_finite_diff(f(x).diff(x), points=dx(x), x0=x-h)
+    -f(-h + x - dx(-h + x)/2)/dx(-h + x) + f(-h + x + dx(-h + x)/2)/dx(-h + x)
+
```

Comment:
I see this function is has public version in `Derivative.as_finite_diff`, and documentation updates may better go there.
I think the documentation here may completely be removed because this mostly contains duplicate information, with the only difference of using deprecated feature in examples.

Reply:
OK, I see your point I will put documentation and example in functions.py.

Should I remove the documentation and keep the example in ``as_finite_diff`` function or remove both?

Reply:
Updates should go to `as_finite_difference` in `sympy/core/functions.py`.
I think that this doc can be removed because it is private function, and would set the docstring of the deprecated function `as_finite_diff.__doc__` as something like 
`"""Deprecated function. Use Diff.as_finite_difference instead"""` so that the sympy website no longer show another misguided doc in https://docs.sympy.org/latest/modules/calculus/index.html?highlight=as_finite_diff#sympy.calculus.finite_diff.as_finite_diff

Reply:
Sounds good. I changed it.